tools/src/bin/crossref.rs

```diff
@@ -61,6 +64,28 @@ fn split_scopes(id: &str) -> Vec<String> {
     return result;
 }
 
+struct StringIntern {
+    set: HashMap<Rc<String>, ()>
```

Comment:
nit: Isn't this basically a `HashSet<Rc<String>>`?

Reply:
As far as I can tell, HashSet<> doesn't support the Entry interface, or anything like that. I want to use the existing value if there is one, insert our new value if there isn't, and avoid repeated lookups. I guess that's overoptimizing. I could use a HashSet and just do two lookups.

Reply:
Well, that's fair... Sounds ok :)